File: Algorithms/Warmup/Manasa and Stones/mns.c
#include <stdio.h>
void manasaAndStones(int n, int a, int b) {
	int i = 0;
	long curr = 0,prev = 0;
	while(i<=n) {
		curr = (a*i) + (b*(n-i));
		if(curr!=prev)
			printf("%ld ",curr);
		prev = curr;
		i++;
	}
	printf("\n");
}
int main() {
	int T,n,a,b;
	scanf("%d",&T);
	while(T--) {
		scanf("%d%d%d",&n,&a,&b);
		if(a<b)
			manasaAndStones(n-1,b,a);
		else
			manasaAndStones(n-1,a,b);
	}
	return 0;
}

File: Algorithms/Warmup/Manasa and Stones/mns.py
def manasaAnsStones(n,a,b):
	i = 0
	prev = 0
	while i<=n:
		curr = (a*i)+(b*(n-i))
		if prev!=curr:
			print curr,
		prev = curr
		i+=1

def __main__():
	T = input()
	for _ in range(0,T):
		n,a,b = input(),input(),input()
		if a<b:
			manasaAnsStones(n-1,b,a)
		else:
			manasaAnsStones(n-1,a,b)
		print ""

if __name__=="__main__":
	__main__()